src/update-process-env.js

```diff
@@ -64,7 +64,32 @@ async function getEnvFromShell (env) {
   }
 
   let {stdout, error} = await new Promise((resolve) => {
-    childProcess.execFile(env.SHELL, ['-ilc', 'command env'], {encoding: 'utf8', timeout: 5000}, (error, stdout) => {
+    let cp
+    let error
+    let stdout = ''
+    const killer = () => {
+      if (cp) {
+        cp.kill()
+      }
+    }
+    process.once('exit', killer)
+
+    cp = childProcess.spawn(env.SHELL, ['-ilc', 'command env'], {encoding: 'utf8', timeout: 5000, detached: true, stdio: ['ignore', 'pipe', process.stderr]})
+
+    const buffers = []
+    cp.on('error', (e) => {
+      error = e
+    })
+    cp.stdout.on('data', (data) => {
+      buffers.push(data)
+    })
+    cp.on('close', (code, signal) => {
+      process.removeListener('exit', killer)
+      if (!buffers || !buffers.length) {
```

Comment:
Surely this can be reduced to
``` js
if (!buffers.length) {
```
since you are explicitly initializing the array a few lines above?